### PR TITLE
TimePicker: Fixes issues with "Recently used absolute ranges" section

### LINKED ISSUE #65756: Recently used absolute ranges is recording nonsense
**What happened**:  The entries in "Recently used absolute ranges" are incorrect in my case/timezone. Looks like date format is not localized there.

**What you expected to happen**:  When I select a new time range with the Timepicker it should show in the "Recently used absolute ranges" so I can quickly reselect it later.

**How to reproduce it (as minimally and precisely as possible)**: select a timerange with the "from" and "to" selector and then click "Apply time range" then the entry added in the section "recently used absolute ranges" is unrelated to the raneg selected.

**Anything else we need to know?**:

**Environment**:

- Grafana version: 9.4.7
- Data source type & version: irrelevent
- OS Grafana is installed on: archlinux
- User OS & Browser:  Windows 10 with Chrome up to date
- Grafana plugins: non
- Others:  Timezone = France, CEST

### PR DESCRIPTION
**Pull request by JoaoSilvaGrafana**
**What is this feature?**

Someone raised an issue where the "Recently used absolute ranges" was giving them nonsensical values for what they have selected:

229301082-4512ba23-a1b4-407c-beb9-b0c961a28c14

This led us to realise that if you have any other date format than the default one (for instance by setting `full_date = DD-MM-YYYY HH:mm:ss` in the `custom.ini` file) this will happen. Because the format that we save in local storage for this section is still 'YYYY-MM-DD', so it is trying to read a date in that format with the configured one, hence the nonsensical numbers. To fix this we pass in the default date format in order to tell the date parser that that is the format of the current date.

While investigating this we also found an issue with the timezone. Similarly to above, it was assuming that the value stored in local storage is in the selected timezone but it is already in 'utc', so we have to pass that to the parser as well.

**Why do we need this feature?**

To be more confident that the Time picker history section works correctly.

**Who is this feature for?**

Anyone that uses the history section of the time picker

**Which issue(s) does this PR fix?**:

Fixes #65756

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: public/app/core/components/TimePicker/TimePickerWithHistory.tsx [hunk 1]
```diff
@@ -41,8 +41,9 @@ export const TimePickerWithHistory = (props: Props) => {
   );
 };
 
-function deserializeHistory(values: TimePickerHistoryItem[], timeZone: TimeZone | undefined): TimeRange[] {
-  return values.map((item) => rangeUtil.convertRawToRange(item, timeZone));
+function deserializeHistory(values: TimePickerHistoryItem[]): TimeRange[] {
+  // The history is saved in UTC and with the default date format, so we need to pass those values to the convertRawToRange
```
Comment (tskarhed) [inline public/app/core/components/TimePicker/TimePickerWithHistory.tsx:45]: Great comment! :+1:

**Review by tskarhed — CHANGES_REQUESTED** [review]
Some tests that cover both the format and timezone correctness would be good. Maybe we can try to do that together? Otherwise I could have a crack at it by myself.

**Review by JoaoSilvaGrafana — COMMENTED** [review]
Nice job with the tests! Just some comment changes, otherwise lgtm!

**Comment by tskarhed** [comment]
I am just trying to finish up with two test cases for TestPickerWithHistory. Good to have it covered, and I get some more test writing exercise :stuck_out_tongue:

**Review by tskarhed — APPROVED** [review]

File: public/app/core/components/TimePicker/TimePickerWithHistory.test.tsx [hunk 2]
```diff
@@ -41,7 +41,10 @@ describe('TimePickerWithHistory', () => {
     onZoom: () => {},
   };
 
-  afterEach(() => window.localStorage.clear());
+  afterEach(() => {
+    window.localStorage.clear();
+    //systemDateFormats.useBrowserLocale();
```
Comment (JoaoSilvaGrafana) [inline public/app/core/components/TimePicker/TimePickerWithHistory.test.tsx:46]:
```suggestion
```

**Pull request merged**